com.unity.render-pipelines.universal/Editor/UniversalRenderPipelineCameraEditor.cs

```diff
@@ -311,12 +311,55 @@ void DrawElementCallback(Rect rect, int index, bool isActive, bool isFocused)
             }
         }
 
+        // Modified version of StageHandle.FindComponentsOfType<T>()
+        // This version more closely represents unity object referencing restrictions.
+        // I added these restrictions:
+        // - Can not reference scene object outside scene
+        // - Can not reference cross scenes
+        // - Can reference child objects if it is prefab
+        Camera[] FindCamerasToReference(GameObject gameObject)
```

Comment:
Based on above comment. Shouldn't this then be `FindTypeToReference<Camera>(GameObject ... )`?
Also is this really a correct place? Shouldn't we have a common implementation?

Reply:
Well I was also considering this, but I decided not to do it for few reasons:
- This function is mostly tailored for finding cameras for camera stacking criteria (etc. we can not include camera prefab in scene).
- This function is only used once.

Now about template point. As this function is mainly tailored for camera stacking adding, for this reason there is no point in my opinion to make it work with other types.